irc/halloy/Portfile

```diff
@@ -133,54 +154,90 @@ cargo.crates \
     cosmic-text                     0.12.1  59fd57d82eb4bfe7ffa9b1cec0c05e2fd378155b47f255a67983cb4afe0e80c2 \
     cpal                            0.15.3  873dab07c8f743075e57f524c583985fbaf745602acbe916a01539364369a779 \
-    cpufeatures                     0.2.14  608697df725056feaccfa42cffdaeeec3fccc4ffc38358ecd19b243e716a78e0 \
+    cpufeatures                     0.2.16  16b80225097f2e5ae4e7179dd2266824648f3e2f49d9134d584b76389d31c4c3 \
     crc32fast                        1.4.2  a97769d94ddab943e4510d138150169a2758b5ef3eb191a9ee688de3e23ef7b3 \
-    crossbeam-deque                  0.8.5  613f8cc01fe9cf1a3eb3d7f488fd2fa8388403e97039e2f73692932e291a770d \
+    crossbeam-deque                  0.8.6  9dd111b7b7f7d55b72c0a6ae361660ee5853c9af73f70c3c2ef6858b950e2e51 \
     crossbeam-epoch                 0.9.18  5b82ac4a3c2ca9c3460964f020e1402edd5753411d7737aa39c3714ad1b5420e \
-    crossbeam-utils                 0.8.20  22ec99545bb0ed0ea7bb9b8e1e9122ea386ff8a48c0922e43f36d45ab09e0e80 \
+    crossbeam-queue                 0.3.12  0f58bbc28f91df819d0aa2a2c00cd19754769c2fad90579b3592b1c9ba7a3115 \
+    crossbeam-utils                 0.8.21  d0a5c400df2834b80a4c3327b3aad3a4c4cd4de0629063962b03235697506a28 \
     crunchy                          0.2.2  7a81dae078cea95a014a339291cec439d2f232ebe854a9d672b796c6afafa9b7 \
+    crypto-bigint                    0.5.5  0dc92fb57ca44df6db8059111ab3af99a63d5d0f8375d9972e319a379c6bab76 \
     crypto-common                    0.1.6  1bfb12502f3fc46cca1bb51ac28df9d618d813cdc3d2f25b9fe775a34af26bb3 \
     ctor-lite                        0.1.0  1f791803201ab277ace03903de1594460708d2d54df6053f2d9e82f592b19e3b \
+    ctr                              0.9.2  0369ee1ad671834580515889b80f2ea915f23b8be8d0daa4bbaf2ac5c7590835 \
     cursor-icon                      1.1.0  96a6ac251f4a2aca6b3f91340350eab87ae57c3f127ffeb585e92bd336717991 \
-    d3d12                           22.0.0  bdbd1f579714e3c809ebd822c81ef148b1ceaeb3d535352afc73fd0c4c6a0017 \
+    curve25519-dalek                 4.1.3  97fb8b7c4503de7d6ae7b42ab72a5a59857b4c937ec27a3d4539dba95b5ab2be \
+    curve25519-dalek-derive          0.1.1  f46882e17999c6cc590af592290432be3bce0428cb0d5f8b6715e4dc7b383eb3 \
+    darling                         0.14.4  7b750cb3417fd1b327431a470f388520309479ab0bf5e323505daf0290cd3850 \
+    darling                        0.20.10  6f63b86c8a8826a49b8c21f08a2d07338eec8d900540f8630dc76284be802989 \
+    darling_core                    0.14.4  109c1ca6e6b7f82cc233a97004ea8ed7ca123a9af07a8230878fcfda9b158bf0 \
+    darling_core                   0.20.10  95133861a8032aaea082871032f5815eb9e98cef03fa916ab4500513994df9e5 \
+    darling_macro                   0.14.4  a4aab4dbc9f7611d8b55048a3a16d2d010c2c8334e46304b40ac1cc14bf3b48e \
+    darling_macro                  0.20.10  d336a2a514f6ccccaa3e09b02d41d35330c07ddf03a62165fcec10bb561c7806 \
```

Comment:
there seem to be duplicates in the list here - is that intentional?

Reply:
This list is generated automatically.

Reply:
So the `cargo.crates` list is generated from a Rust project's `Cargo.lock` file (which itself derives from the `Cargo.toml` file).

It's quite common to see more than one version of the same crate in a Rust project's dependencies list.

This is possible for a number of reasons, one of them being that some other dependency crates may each themselves use different versions of the same root level crate.

You can see entries for the 2 versions of `darline_core` [here](https://github.com/squidowl/halloy/blob/a1b6dca61a6d74ab95551474d2e468f36bc96390/Cargo.lock#L1394) and [here](https://github.com/squidowl/halloy/blob/a1b6dca61a6d74ab95551474d2e468f36bc96390/Cargo.lock#L1408).

So I think we're fine here.